File: 24 - Sort Array by Parity/solution.js
function sortArrayByParity(nums) {
  // Your code here
  let evens = [];
  let odds = [];

  for (let num of nums) {
    if (num % 2 === 0) {
      evens.push(num);
    } else {
      odds.push(num);
    }
  }

  return [...evens, ...odds];
}

//   Test cases
console.log(sortArrayByParity([3, 1, 2, 4]));

File: 24 - Sort Array by Parity/solution.py
def sort_array_by_parity(nums):
    # Your code here
    evens = []
    odds = []

    for num in nums:
        if num % 2 == 0:
            evens.append(num)
        else:
            odds.append(num)

    return evens + odds


#   Test cases
print(sort_array_by_parity([3, 1, 2, 4]))